haskell-best-practices.md

```diff
@@ -396,6 +396,37 @@ f (ThingA a) = a == 0
 f (ThingB b) = if b then 1 else 2
 ```
 
+## Avoid `OverloadedLists` and `MonoTraversable`
+
+`OverloadedLists` work from a typelcass `IsList`:
+
+```
+class IsList l where
+type Item l
+fromList  :: [Item l] -> l
+toList    :: l -> [Item l]
+```
+  
+The `mono-traversable` package defines a series typeclasses:
+
+```
+type family Element mono
+
+class MonoFoldable mono where
+ofoldMap :: Monoid m => (Element mono -> m) -> mono -> m 
+...
+```
+  
+You can see that both of these typeclasses utilize type families. `IsList` has an associated type `Item` and `MonoFoldable` uses an open type family `Element`. They are nearly identical in their utilization of these techniques. I'll focus mainly on `mono-traversable`, but arguments against it are very transferable to `OverloadedLists`.
```

Comment:
I'd love to see these hard-wrapped as we do *mostly* consistently in our markdown (this document isn't a great example). But I get that it's a pain without a properly-configured editor, so consider this non-blocking.